### PR TITLE
Support rules_go

### PR DESCRIPTION
**Pull request by cerisier**
Base for implementation of #45 

The first issue we had was linked to duplicated linker args produced by the go linker driver:

```
ld.lld: error: duplicate symbol: __dso_handle
>>> defined at crtbegin.c
>>>            crtbegin.o:(__dso_handle) in archive bazel-out/linux_x86_64-fastbuild-ST-ca5ab100789a/bin/external/toolchains_llvm_bootstrapped++http_archive+compiler-rt/clang_rt.crtbegin.static_/libclang_rt.crtbegin.static.a
>>> defined at crtbegin.c
>>>            crtbegin.o:(.data+0x0) in archive bazel-out/linux_x86_64-fastbuild-ST-ca5ab100789a/bin/external/toolchains_llvm_bootstrapped++http_archive+compiler-rt/clang_rt.crtbegin.static_/libclang_rt.crtbegin.static.a
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

because CRTs are being passed multiple times by `go tool link`.

The next issue is described below...

### CODE REVIEW

File: examples/rules_go/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    visibility = ["//visibility:public"],
+)
+
+go_cross_binary(
+    name = "cgo_linux_amd64",
+    target = "cgo",
+    platform = "@toolchains_llvm_bootstrapped//platforms:linux_x86_64",
+)
+
+build_test(
```
Comment (dzbarsky) [inline examples/rules_go/BUILD.bazel]: nit: you don't really need `build_test`, `bazel test` will already ensure it builds. The only edge case is if you have no test targets, then `bazel test` will fail even if they all compile. But we do have a `go_test` here, so we're good

File: examples/rules_go/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    name = "hermetic_cc_toolchain_bcr_tests",
+    version = "0.1.0",
+)
+
+bazel_dep(name = "platforms", version = "1.0.0")
+bazel_dep(name = "rules_go", version = "0.59.0")
+bazel_dep(name = "bazel_skylib", version = "1.8.2")
+
+# https://github.com/cerisier/toolchains_llvm_bootstrapped/releases
+bazel_dep(name = "toolchains_llvm_bootstrapped", version = "0.0.0", dev_dependency = True)
```
Comment (dzbarsky) [inline examples/rules_go/MODULE.bazel]: you can snip `dev_dependency` here and below, it doesn't really matter :)

File: examples/rules_go/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    srcs = ["cgo_test.go"],
+    embed = [":cgo_lib"],
+)
+
+go_binary(
+    name = "cgo",
+    embed = [":cgo_lib"],
+    gc_linkopts = select({
+        "@platforms//os:macos": [
+            "-w",  # https://github.com/ziglang/zig/issues/15439
```
Comment (dzbarsky) [inline examples/rules_go/BUILD.bazel]: are these zig issues relevant?

File: examples/rules_go/BUILD.bazel (hunk 1 above)
Comment (cerisier) [reply examples/rules_go/BUILD.bazel]: Even if'm testing the cross compiled target ? the go_test seems to only test for exec.

File: e2e/rules_go/cgo.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package main
```
Comment (dzbarsky) [inline e2e/rules_go/cgo.go:1]: (above) remove `a.out` from git

**Review by dzbarsky — APPROVED** [review]

File: examples/rules_go/BUILD.bazel (hunk 3 above)
Comment (cerisier) [reply examples/rules_go/BUILD.bazel]: Not for us, removed.

File: examples/rules_go/BUILD.bazel (hunk 1 above)
Comment (dzbarsky) [reply examples/rules_go/BUILD.bazel]: yeah if you do `bazel test //...` in this module it will still build the binaries. You can double-check by adding a syntax error and you'll see it fails, even if the `go_test` target is commented out.

not a huge deal either way though!
